src/cpp/core/include/core/algorithm/Set.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Set.hpp
+ *
+ * Copyright (C) 2009-2015 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+#ifndef CORE_ALGORITHM_SET_HPP
+#define CORE_ALGORITHM_SET_HPP
+
+// Convenience algorithms -- not necessarily as fast as just
+// using the STL + iterators, but often easier to use.
+
+#include <algorithm>
+
+namespace rstudio {
+namespace core {
+namespace algorithm {
+
+#define RS_SET_OPERATION(__OPERATION__)                                        \
```

Comment:
In this particular bit of code I'm just trying to be as DRY as possible, since the `set_*` functions I'm defining just dispatch down to the appropriate calls to `std::set_*` (with the appropriate 'set up' done on each container). At least here, I think it's relatively clear (although could be less debuggable).

I agree that `IF_ERROR` is not worth the mental load though so I'll go ahead and strip that out.